Description: Fix arm/i386/mipsel build.
 ffmpegfs (2.10-2) unstable; urgency=medium
 .
   * Drop MULTIFRAME_NODE typedef in favour of auto to avoid build fails under
     arm/i386/mipsel.
Author: Norbert Schlia <nschlia@oblivion-software.de>

---
The information above should follow the Patch Tagging Guidelines, please
checkout http://dep.debian.net/deps/dep3/ to learn about the format. Here
are templates for supplementary fields that you might want to add:

Origin: <vendor|upstream|other>, <url of original patch>
Bug: <url in upstream bugtracker>
Bug-Debian: https://bugs.debian.org/<bugnumber>
Bug-Ubuntu: https://launchpad.net/bugs/<bugnumber>
Forwarded: <no|not-needed|url proving that it has been forwarded>
Reviewed-By: <name and email of someone who approved the patch>
Last-Update: 2022-05-15

--- ffmpegfs-2.10.orig/src/ffmpeg_transcoder.cc
+++ ffmpegfs-2.10/src/ffmpeg_transcoder.cc
@@ -5394,7 +5394,10 @@ int FFmpeg_Transcoder::process_single_fr
                 // Frame sets: no audio, no subtitles, no output stream (index)
                 while (!m_frame_map.empty())
                 {
-                    MULTIFRAME_NODE nh = m_frame_map.extract(m_frame_map.cbegin());
+                    // Although I abhore "auto", it seems that nodes are declared slightly different throughout
+                    // serveral std::map implementations. To avoid #ifdef orgies, I resign: Using "auto" here
+                    // is much more coherent.
+                    auto nh = m_frame_map.extract(m_frame_map.cbegin());
                     const MULTIFRAME & multiframe = nh.mapped();
                     int ret = 0;
                     int data_written = 0;
--- ffmpegfs-2.10.orig/src/ffmpeg_transcoder.h
+++ ffmpegfs-2.10/src/ffmpeg_transcoder.h
@@ -169,15 +169,6 @@ protected:
     } HWACCELMODE;
 
     typedef std::variant<FFmpeg_Frame, FFmpeg_Subtitle> MULTIFRAME; /**< @brief Combined audio/videoframe and subtitle */
-    /**
-     * @brief MULTIFRAME_NODE node_type for use with std::map<...>::extract.
-     * To be honest I was incredibly close to using "auto" as in all examples,
-     * although abhoring this keyword.
-     * This syntax is really credible for the ugliest piece of C++ code of
-     * all times (grin).
-     */
-    typedef std::multimap<long, MULTIFRAME, std::less<long>, std::allocator<std::pair<const long, MULTIFRAME>>>::node_type MULTIFRAME_NODE;
-
     typedef std::multimap<int64_t, MULTIFRAME>  MULTIFRAME_MAP;     /**< @brief Audio frame/video frame/subtitle buffer */
     typedef std::map<int, int>                  STREAM_MAP;         /**< @brief Map input subtitle stream to output stream */
 
